uber/site_sections/registration.py

```diff
@@ -417,74 +413,76 @@ def take_back_merch(self, session, id):
         session.commit()
         return '{a.full_name} ({a.badge}) merch handout canceled'.format(a=attendee)
 
-    if AT_THE_CON or DEV_BOX:
-        @unrestricted
-        def register(self, session, message='', **params):
-            params['id'] = 'None'
-            attendee = session.attendee(params, bools=['international'], checkgroups=['interests'], restricted=True, ignore_csrf=True)
-            if 'first_name' in params:
-                if not attendee.payment_method:
-                    message = 'Please select a payment type'
-                elif not attendee.first_name or not attendee.last_name:
-                    message = 'First and Last Name are required fields'
-                elif attendee.ec_phone[:1] != '+' and not attendee.international and len(re.compile('[0-9]').findall(attendee.ec_phone)) != 10:
-                    message = 'Enter a 10-digit emergency contact number'
-                elif re.search(SAME_NUMBER_REPEATED, re.sub(r'[^0-9]', '', attendee.ec_phone)):
-                    message = 'Please enter a real emergency contact number'
-                elif not attendee.age_group:
-                    message = 'Please select an age category'
-                elif attendee.payment_method == MANUAL and not re.match(EMAIL_RE, attendee.email):
-                    message = 'Email address is required to pay with a credit card at our registration desk'
-                elif attendee.badge_type not in [ATTENDEE_BADGE, ONE_DAY_BADGE]:
-                    message = 'No hacking allowed!'
-                else:
-                    session.add(attendee)
-                    attendee.badge_num = 0
-                    if not attendee.zip_code:
-                        attendee.zip_code = '00000'
-                    attendee.save()
-                    message = 'Thanks!  Please queue in the {} line and have your photo ID and {} ready.'
-                    if attendee.payment_method == STRIPE:
-                        raise HTTPRedirect('pay?id={}', attendee.id)
-                    elif attendee.payment_method == GROUP:
-                        message = 'Please proceed to the preregistration line to pick up your badge.'
-                        attendee.paid = PAID_BY_GROUP
-                    elif attendee.payment_method == CASH:
-                        message = message.format('cash', '${}'.format(attendee.total_cost))
-                    elif attendee.payment_method == MANUAL:
-                        message = message.format('credit card', 'credit card')
-                    raise HTTPRedirect('register?message={}', message)
+    @unrestricted
+    @check_atd
+    def register(self, session, message='', **params):
+        params['id'] = 'None'
+        attendee = session.attendee(params, restricted=True, ignore_csrf=True)
+        if 'first_name' in params:
```

Comment:
Again, with the new validations being nice and extensible, I'm not keen on leaving these hardcoded. Though if we simply unify the pre-reg and at-door pages first, that'll fix this by default.